### PR TITLE
Trivial: fix references to share/rpcuser (now share/rpcauth)

### PR DESCRIPTION
**Pull request by masonicboom**
Commit 3fdb297 renamed share/rpcuser to share/rpcauth but left references to the old path in code; this commit fixes the old references.

```
-BEGIN VERIFY SCRIPT-
git grep --files-with-matches 'share/rpcuser' src/*.cpp | xargs sed -i -E 's:share/rpcuser:share/rpcauth:g'
git grep --files-with-matches 'share/rpcuser' test/functional/*.py | xargs sed -i -E 's:share/rpcuser:share/rpcauth:g'
-END VERIFY SCRIPT-
```

### CODE REVIEW

**Comment by masonicboom** [comment]
@hkjn got time for a review? This is just a little cleanup from an old change of yours. Also tagging @laanwj who merged that one.

**Comment by sipa** [comment]
Can this be implemented as a scripted diff?

File: doc/man/bitcoin-qt.1 [hunk 1]
```diff
@@ -514,7 +514,7 @@ Password for JSON\-RPC connections
 .IP
 Username and hashed password for JSON\-RPC connections. The field
 <userpw> comes in the format: <USERNAME>:<SALT>$<HASH>. A
-canonical python script is included in share/rpcuser. The client
+canonical python script is included in share/rpcauth. The client
```
Comment (MarcoFalke) [inline doc/man/bitcoin-qt.1]: .\" DO NOT MODIFY THIS FILE!  It was generated by help2man 1.47.3.

File: src/qt/bitcoinstrings.cpp [hunk 2]
```diff
@@ -231,7 +231,7 @@ QT_TRANSLATE_NOOP("bitcoin-core", ""
 QT_TRANSLATE_NOOP("bitcoin-core", ""
 "Username and hashed password for JSON-RPC connections. The field <userpw> "
 "comes in the format: <USERNAME>:<SALT>$<HASH>. A canonical python script is "
-"included in share/rpcuser. The client then connects normally using the "
+"included in share/rpcauth. The client then connects normally using the "
```
Comment (MarcoFalke) [inline src/qt/bitcoinstrings.cpp]: // Automatically generated by extract_strings_qt.py

File: doc/man/bitcoin-qt.1 (hunk 1 above)
Comment (masonicboom) [reply doc/man/bitcoin-qt.1]: Fixing

File: src/qt/bitcoinstrings.cpp (hunk 2 above)
Comment (masonicboom) [reply src/qt/bitcoinstrings.cpp]: Fixing

**Comment by masonicboom** [comment]
Thanks for review. @MarcoFalke I made the fixes you mentioned. @sipa it’s now a scripted diff.

**Comment by fanquake** [comment]
utACK c38d567

The only instances of `share/rpcuser` left in the code are in old release-notes, `doc/man` and `bitcoinstrings.cpp`

I also removed the reference to codemod from the PR description, as that isn't required.

**Comment by MarcoFalke** [comment]
utACK c38d567cbf25495fea7e941c75b3603e79d58061. Just noting that the scripted diff also modifies files that are not tracked by git. So I'd prefer to not do this 3-line change as a scripted diff, but I don't care either way.

**Comment by masonicboom** [comment]
Ok. I restricted the script to only touch files in version control.

If someone could point me to a reference on the scripted diff concept I would appreciate it. I'm curious what it buys us in this case.

Here's a test plan:
1. pull changes
2. build
3. `src/bitcoind --help | grep "share/rpcuser"` comes up empty.
4. `src/bitcoind --help | grep "share/rpcauth"` produces:

       canonical python script is included in share/rpcauth. The client

**Pull request merged**